rustup-init.sh

```diff
@@ -72,7 +72,7 @@ main() {
     local _url="${RUSTUP_UPDATE_ROOT}/dist/${_arch}/rustup-init${_ext}"
 
     local _dir
-    _dir="$(mktemp -d 2>/dev/null || ensure mktemp -d -t rustup)"
+    _dir="$(mktemp -d 2>/dev/null || ensure mktemp -d -t rustup-XXX)"
```

Comment:
lets just remove the || path here - even OpenBSD mktemp supports -d as-is. https://man.openbsd.org/mktemp.1 : there's no possible case I can see where we're making things better with this.

#183 is the bug that this was meant to fix initially. mktemp on OS X - https://github.com/rust-lang/rustup/commit/7719d9143bfeb9dd9e719a07a0c3e2b3d3ad8732

Now, here's the man page for OS X mktemp: https://ss64.com/osx/mktemp.html

again, -d is supported.

So, I don't know why it was failing for rustup-setup as reported in #183, but the fix was not correctly diagnosed: the original command was valid, and the fix was invalid: note the language from the man page "The template may be any file name with some number of `Xs' appended to it, for example /tmp/temp.XXXX. ".

So - lets:
* go back to a single mktemp -d
* remove 2>/dev/null so that we can actually find out what goes wrong in the long tail here and fix the real bug in future

Reply:
This makes a lot of sense.  I wasn't in the mood to go digging in other OS manpages, so thanks for that Robert.

Reply:
You're welcome :)

Reply:
Done

Reply:
We want
```
local _dir="$(ensure mktemp -d)"
```

Reply:
Right, I forgot to stage my changes before committing... should be good now